docs/guide/counter/plugins/counter/counter.go

```diff
@@ -1,98 +1,197 @@
 package counter
 
 import (
-	"fmt"
+	rawerr "errors"
 
 	abci "github.com/tendermint/abci/types"
-	"github.com/tendermint/basecoin/types"
 	"github.com/tendermint/go-wire"
+
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/errors"
+	"github.com/tendermint/basecoin/modules/coin"
+	"github.com/tendermint/basecoin/stack"
+	"github.com/tendermint/basecoin/types"
 )
 
-type CounterPluginState struct {
-	Counter   int
-	TotalFees types.Coins
+// CounterTx
+//--------------------------------------------------------------------------------
+
+// register the tx type with it's validation logic
+// make sure to use the name of the handler as the prefix in the tx type,
+// so it gets routed properly
+const (
+	NameCounter = "cntr"
+	ByteTx      = 0x21
```

Comment:
Please clarify. I define a constant that is used to register with go- wire. Should I change the name? This is how go- wire registration works in go-crypto, so if there is a problem, we need to change it everywhere.